consumer/app/artifacts.py

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2019 by eHealth Africa : http://www.eHealthAfrica.org
+#
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import fnmatch
+import requests
+from time import sleep
+from typing import (
+    Callable,
+    List
+)
+from uuid import uuid4
+
+from confluent_kafka import KafkaException
+
+# Consumer SDK
+from aet.exceptions import ConsumerHttpException
+from aet.job import BaseJob, JobStatus
+from aet.kafka import KafkaConsumer, FilterConfig, MaskConfig
+from aet.logger import callback_logger, get_logger
+from aet.resource import BaseResource, lock
+
+from app.config import get_kafka_config
+from app.fixtures import schemas
+
+
+LOG = get_logger('artifacts')
+KAFKA_CONFIG = get_kafka_config()
+
+
+class CKANInstance(BaseResource):
+    schema = schemas.CKAN_INSTANCE
+    jobs_path = '$.ckan'
+    name = 'ckan'
+    public_actions = BaseResource.public_actions + [
+        'test_connection'
+    ]
+    session: requests.Session() = None
+
+    def get_session(self):
+        if self.session:
+            return self.session
+        self.session = requests.Session()
+        try:
+            self.session.auth = (
+                self.definition.user,
+                self.definition.password
+            )
+        except AttributeError:
+            pass  # may not need creds
+        # add an _id so we can check the instance
+        setattr(self.session, 'instance_id', str(uuid4()))
+        return self.session
+
+    @lock
```

Comment:
We only need to @lock this method if having multiple requests at the same time isn't safe to do in parallel. I've probably overused it in the past, but something to remember.